executor/instruction/has_executor.rs

```diff
@@ -131,6 +134,7 @@ impl HasExecutor {
         };
         Ok(Self {
             has,
+            sort_mode,
```

Comment:
I should delete the `iterate_mode` and `filter_fn` from the state, as they're dynamic too now.